exchanges/btcmarkets/btcmarkets.go

```diff
@@ -52,316 +83,531 @@ type BTCMarkets struct {
 		&resp)
-	if err != nil {
-		return "", err
-	}
+}
 
-	if !resp.Success {
-		return "", errors.New(resp.ErrorMessage)
-	}
+// ListAssets lists all available assets
+func (b *BTCMarkets) ListAssets() ([]AssetData, error) {
+	var resp []AssetData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsAssets, nil, &resp)
+}
 
-	return resp.Status, nil
+// GetTransactions gets trading fees
+func (b *BTCMarkets) GetTransactions(assetName string) ([]TransactionData, error) {
+	var resp []TransactionData
+	req := make(map[string]interface{})
+	req["assetName"] = assetName
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsTransactions, req, &resp)
 }
 
-// WithdrawAUD withdraws AUD into a designated bank address
-// Does not return a TxID!
-func (b *BTCMarkets) WithdrawAUD(accountName, accountNumber, bankName, bsbNumber string, amount float64) (string, error) {
-	newAmount := int64(amount * float64(common.SatoshisPerBTC))
+// CreateNewReport creates a new report
+func (b *BTCMarkets) CreateNewReport(reportType, format string) (CreateReportResp, error) {
+	var resp CreateReportResp
+	req := make(map[string]interface{})
+	req["type"] = reportType
+	req["format"] = format
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsReports, req, &resp)
+}
 
-	req := WithdrawRequestAUD{
-		AccountName:   accountName,
-		AccountNumber: accountNumber,
-		BankName:      bankName,
-		BSBNumber:     bsbNumber,
-		Amount:        newAmount,
-		Currency:      "AUD",
-	}
+// GetReport finds details bout a past report
+func (b *BTCMarkets) GetReport(reportID string) (ReportData, error) {
+	var resp ReportData
+	return resp, b.SendAuthenticatedRequest(http.MethodGet, btcMarketsReports+"/"+reportID,
+		nil, &resp)
+}
 
-	resp := Response{}
-	err := b.SendAuthenticatedRequest(http.MethodPost, btcMarketsWithdrawAud,
-		req,
-		&resp)
-	if err != nil {
-		return "", err
+// RequestWithdraw requests withdrawals
+func (b *BTCMarkets) RequestWithdraw(assetName string, amount float64,
+	toAddress, accountName, accountNumber, bsbNumber, bankName string) (TransferData, error) {
+	var resp TransferData
+	req := make(map[string]interface{})
+	req["assetName"] = assetName
+	req["amount"] = strconv.FormatFloat(amount, 'f', -1, 64)
+	if assetName != "AUD" {
+		req["toAddress"] = toAddress
+	} else {
+		if accountName != "" {
+			req["accountName"] = accountName
+		}
+		if accountNumber != "" {
+			req["accountNumber"] = accountNumber
+		}
+		if bsbNumber != "" {
+			req["bsbNumber"] = bsbNumber
+		}
+		if bankName != "" {
+			req["bankName"] = bankName
+		}
 	}
+	return resp, b.SendAuthenticatedRequest(http.MethodPost, btcMarketsWithdrawals, req, &resp)
+}
 
-	if !resp.Success {
-		return "", errors.New(resp.ErrorMessage)
+// BatchPlaceCancelOrders places and cancels batch orders
+func (b *BTCMarkets) BatchPlaceCancelOrders(cancelOrders []CancelOrderMethod, placeOrders []PlaceOrderMethod) (BatchPlaceCancelResponse, error) {
```

Comment:
revert to cancel batch and place batch slice types because its annoying doing the work in higher calls, then just wrap the cancel order or place order methods over the data when we range.